packages/vscode-extension/src/commands/ImportObjects.ts

```diff
@@ -0,0 +1,247 @@
+			throw 'Unexpected error at list objects';
+		}
+
+		const actionResult = await this.customObjectService.importObjects(
+			relativeDestinationFolder,
+			appId,
+			selectedScriptIds,
+			includeReferencedFiles === this.translationService.getMessage(ANSWERS.YES),
+			this.executionPath,
+			this.vsConsoleLogger
+		);
+
+		this.showOutput(actionResult);
+		return actionResult;
+	}
+
+	private getDestinationFolder(pathDir: string) {
+		const isDirectory = fs.lstatSync(pathDir).isDirectory();
+		const directoryName = isDirectory ? pathDir : path.dirname(pathDir);
+		const destinationFolder = directoryName.split(this.getProjectFolderPath())[1].replace(/\\/gi, '/');
+		return destinationFolder;
+	}
+
+	private async promptFilterAppId() {
+		return await window.showQuickPick([this.translationService.getMessage(ANSWERS.YES), this.translationService.getMessage(ANSWERS.NO)], {
+			placeHolder: this.translationService.getMessage(IMPORT_OBJECTS.QUESTIONS.FILTER_APP_ID),
+			canPickMany: false,
+		});
+	}
+
+	private async promptAppId(projectInfoService: { getApplicationId: () => string }): Promise<string | undefined> {
+		const defaultAppId = projectInfoService.getApplicationId();
+		let appId = await window.showInputBox({
+			ignoreFocusOut: true,
+			placeHolder: this.translationService.getMessage(IMPORT_OBJECTS.QUESTIONS.APP_ID, defaultAppId),
+			validateInput: (fieldValue) => {
+				if (fieldValue === '') {
+					return null; //In this case empty field is valid
+				}
+				let validationResult = InteractiveAnswersValidator.showValidationResults(fieldValue, InteractiveAnswersValidator.validateSuiteApp);
+				return typeof validationResult === 'string' ? validationResult : null;
+			},
+		});
+		if (appId === undefined) {
+			return;
+		}
+
+		if (appId === '') {
+			appId = defaultAppId;
+		}
+		return appId;
+	}
+
+	private async promptScriptIdFilter() {
+		let scriptId = await window.showInputBox({
+			ignoreFocusOut: true,
+			placeHolder: this.translationService.getMessage(IMPORT_OBJECTS.QUESTIONS.SCRIPT_ID),
+			validateInput: (fieldValue) => {
+				if (fieldValue === '') {
+					return null; //In this case empty field is valid
+				}
+				let validationResult = InteractiveAnswersValidator.showValidationResults(fieldValue, InteractiveAnswersValidator.validateScriptId);
+				return typeof validationResult === 'string' ? validationResult : null;
+			},
+		});
+
+		return scriptId;
+	}
+
+	private async promptObjectTypes(): Promise<string[] | undefined> {
+		return await window.showQuickPick(
+			objectTypes.map((objectType) => objectType.value.type),
+			{
+				placeHolder: this.translationService.getMessage(IMPORT_OBJECTS.QUESTIONS.SELECT_TYPES),
+				canPickMany: true,
+			}
+		);
+	}
+
+	private async listObjects(appId: string, selectedObjectTypes: string[], scriptId: string) {
+		if (!this.executionPath) {
+			//already  checked in validate. Should not throw
+			throw 'Unexpected error at list objects';
+		}
```

Comment:
I would like to move this check to line 25 in the main execute method.
Then I would directly call this.customObjectService.listObjects() from line 63 and delete the whole listObjects() method.